source/base/subscriptor.cc

```diff
@@ -143,16 +146,18 @@ Subscriptor::operator=(const Subscriptor &s)
 Subscriptor &
 Subscriptor::operator=(Subscriptor &&s) noexcept
 {
-  check_no_subscribers();
-  s.check_no_subscribers();
+  for (auto validity_ptr : validity_pointers)
+    *validity_ptr = false;
+  for (auto validity_ptr : s.validity_pointers)
+    *validity_ptr = false;
```

Comment:
I can't quite convince myself why this is correct. Can you explain in a comment here?

Reply:
The object moved from is normally invalidated. Hence, I believe it is a valid assumption to invalidate all the pointers to it.

For the object copied or moved to, the reasoning is a bit more controversial and I went for invalidation for safety. On a second thought, I guess it is fine to say that the object is valid even if it's content has been replaced by something else. 

Does that make sense to you?